src/lib/lib.jl

```diff
@@ -14,6 +14,7 @@ accum(x, y, zs...) = accum(accum(x, y), zs...)
 
 accum(x::Tuple, y::Tuple) = accum.(x, y)
 accum(x::AbstractArray, y::AbstractArray) = accum.(x, y)
+accum(x::DenseArray, y::AbstractArray) = x .= accum.(x, y)
```

Comment:
ChainRulesCore has a more complicated set of rules for working out if this is allowed.
Since it is not clear if all `Dense` subtypes are going to be mutable
Which we can get by doing
```suggestion
accum(x::DenseArray, y::AbstractArray) = ChainRulesCore.add!!(x, y)
```

https://github.com/JuliaDiff/ChainRulesCore.jl/blob/5f1613ffe4f46287984f2fc2733bf4ec8a8d65c9/src/accumulation.jl#L27-L63

Reply:
No strong opinions, this was the low-complication way of catching Array and CuArray. 
```
julia> subtypes(DenseArray)
6-element Vector{Any}:
 Array
 Base.CodeUnits
 Base.Experimental.Const
 Random.UnsafeView
 SharedArrays.SharedArray
 SuiteSparse.CHOLMOD.Dense

julia> Base.Experimental.Const(rand(2)) isa DenseArray
true
```

Reply:
Lets leave it as `DenseArray` for now and worry about it later